src/Service/Configurations/RuntimeConfigProvider.cs

```diff
@@ -5,407 +5,252 @@
+    /// </summary>
+    /// <param name="runtimeConfig">Populated runtime configuration, if present.</param>
+    /// <returns>True when runtime config is provided, otherwise false.</returns>
+    public bool TryGetLoadedConfig([NotNullWhen(true)] out RuntimeConfig? runtimeConfig)
+    {
+        runtimeConfig = _runtimeConfig;
+        return _runtimeConfig is not null;
+    }
 
-            return configLoadSucceeded;
+    /// <summary>
+    /// Initialize the runtime configuration provider with the specified configurations.
+    /// This initialization method is used when the configuration is sent to the ConfigurationController
+    /// in the form of a string instead of reading the configuration from a configuration file.
+    /// This method assumes the connection string is provided as part of the configuration.
+    /// </summary>
+    /// <param name="configuration">The engine configuration.</param>
+    /// <param name="schema">The GraphQL Schema. Can be left null for SQL configurations.</param>
+    /// <param name="accessToken">The string representation of a managed identity access token</param>
+    /// <returns>true if the initialization succeeded, false otherwise.</returns>
+    public async Task<bool> Initialize(
+        string configuration,
+        string? schema,
+        string? accessToken)
+    {
+        if (string.IsNullOrEmpty(configuration))
+        {
+            throw new ArgumentException($"'{nameof(configuration)}' cannot be null or empty.", nameof(configuration));
         }
 
-        /// <summary>
-        /// Initialize the runtime configuration provider with the specified configurations.
-        /// This initialization method is used when the configuration is sent to the ConfigurationController
-        /// in the form of a string instead of reading the configuration from a configuration file.
-        /// </summary>
-        /// <param name="configuration">The engine configuration.</param>
-        /// <param name="schema">The GraphQL Schema. Can be left null for SQL configurations.</param>
-        /// <param name="connectionString">The connection string to the database.</param>
-        /// <param name="accessToken">The string representation of a managed identity access token</param>
-        /// <returns>true if the initialization succeeded, false otherwise.</returns>
-        public async Task<bool> Initialize(
-            string configuration,
-            string? schema,
-            string connectionString,
-            string? accessToken)
+        if (RuntimeConfigLoader.TryParseConfig(
+                configuration,
+                out RuntimeConfig? runtimeConfig))
         {
-            if (string.IsNullOrEmpty(connectionString))
-            {
-                throw new ArgumentException($"'{nameof(connectionString)}' cannot be null or empty.", nameof(connectionString));
-            }
+            _runtimeConfig = runtimeConfig;
```

Comment:
Is the call to `RuntimeConfiguration!.MapGraphQLSingularTypeToEntityName(ConfigProviderLogger);` not needed anymore? I remember this is needed for CosmosDB, How do we handle this now?

Reply:
It's primarily handled when we deserialize the JSON file to the object model. Cosmos does have a slight variation on this where it will also fallback to looking at the `@model` directive but only on an as-needed basis.

Reply:
updated here:
https://github.com/Azure/data-api-builder/blob/04195315ca62c4a3fcc4ffd7e90b57501df91f32/src/Service/Services/MetadataProviders/CosmosSqlMetadataProvider.cs#L308-L344

Reply:
Resolving since there is a mitigation which addresses the initial concern of removing `RuntimeConfiguration!.MapGraphQLSingularTypeToEntityName`. Originally, this was the code for `GetEntityName()` which previously referred to the dictionary populated by `MapGraphQLSingularTypeToEntityName()` 
https://github.com/Azure/data-api-builder/blob/f026fd3825e3314c389fdacd83acdb6d22d30339/src/Service/Services/MetadataProviders/CosmosSqlMetadataProvider.cs#L311C9-L327C10

Now, calls to `GetEntityName()` for Cosmos have the logic i referenced in a previous comment, where the logic looks at the TopLevel entity name, then for when a different GraphQL object type name than the entity name is present, look at the value set in the model directive, then look at the singular name value.

https://github.com/Azure/data-api-builder/blob/04195315ca62c4a3fcc4ffd7e90b57501df91f32/src/Service/Services/MetadataProviders/CosmosSqlMetadataProvider.cs#L308-L344